src/boxatlas/contactstabilization.py

```diff
@@ -246,24 +254,23 @@ def add_constraints(self, initial_state, vlimb_max=5, Mq=10, Mv=100, Mf=1000):
                 self.prog.AddLinearConstraint(self.vars.vlimb[k](self.ts[0])[i] == 0)
                 self.prog.AddLinearConstraint(self.vars.qlimb[k](self.ts[0])[i] == initial_state.qlimb[k][i])
 
-    def add_costs(self):
+    def add_costs(self, desired_state):
         num_limbs = len(self.robot.limb_bounds)
         cost_weights = self.params['costs']
         self.prog.AddQuadraticCost(
             cost_weights['contact_force'] * np.sum(np.sum(np.power(self.vars.contact_force[k](t), 2)) for t in self.ts[:-1] for k in range(num_limbs)))
         self.prog.AddQuadraticCost(
-            cost_weights['qcom_running'] * np.sum(np.sum(np.power(q - np.array([0, 1]), 2)) for q in self.vars.qcom.at_all_breaks()))
-        self.prog.AddQuadraticCost(
-            0.001 * np.sum(np.sum(np.power(self.vars.qcom.derivative().derivative()(t), 2)) for t in self.ts[:-1]))
+            cost_weights['qcom_running'] * np.sum(np.sum(np.power(q - desired_state.qcom, 2)) for q in self.vars.qcom.at_all_breaks()))
+
         self.prog.AddQuadraticCost(
-            0.001 * np.sum(np.sum(np.power(self.vars.qlimb[k].derivative()(t), 2)) for t in self.ts[:-1] for k in range(num_limbs)))
+            0.001 * np.sum([np.sum(np.power(q.derivative()(t), 2) for t in self.ts[:-1]) for q in self.vars.qlimb]))
```

Comment:
could we replace 0.001 with something like `cost_weights['nominal_pose']` so we make sure to keep all the magic weights in one place?